src/meta/mod.rs

```diff
@@ -25,7 +25,7 @@ pub struct Meta {
     pub date: Date,
     pub owner: Owner,
     pub file_type: FileType,
-    pub size: Size,
+    pub size: Option<Size>,
```

Comment:
I think you can remove the Option here. Instead of calling `render()` if there is a Size or printing "-" you can move this logic inside the Size.render_value() which would return an empty String if the it's a directory or returning the actual size. Same for Size.render_unit().